src/Language/Spectacle/Syntax/Logic.hs

```diff
@@ -0,0 +1,159 @@
+-- | Quantifiers and logic.
+--
+-- @since 0.1.0.0
+module Language.Spectacle.Syntax.Logic
+  ( Logic (Logic),
+    Effect (Forall, Exists),
+    forall,
+    exists,
+    complement,
+    conjunct,
+    disjunct,
+    implies,
+    iff,
+    runLogic,
+  )
+where
+
+import Control.Applicative (Applicative (liftA2))
+import Control.Monad (filterM, unless, when)
+import Data.Coerce (coerce)
+import Data.Void (absurd)
+
+import Data.Functor.Loom (hoist, runLoom, (~>~))
+import Language.Spectacle.Exception.RuntimeException
+  ( QuantifierException (ExistsViolated, ForallViolated),
+    RuntimeException (QuantifierException),
+  )
+import Language.Spectacle.Lang
+  ( Effect,
+    Lang (Pure, Yield),
+    Member (projectS),
+    Members,
+    Union (Op, Scoped),
+    decomposeOp,
+    decomposeS,
+    scope,
+  )
+import Language.Spectacle.Syntax.Error (Error, throwE)
+import Language.Spectacle.Syntax.Logic.Internal
+  ( Effect (Complement, Conjunct, Disjunct, Exists, Forall),
+    Logic (Logic),
+  )
+import Language.Spectacle.Syntax.NonDet (NonDet, oneOf)
+
+-- -------------------------------------------------------------------------------------------------
+
+-- | Universally quantify over some foldable container @f a@. A nondeterministically chosen element
+-- in @f a@ will be return so long as the given predicate is 'True' for all elements in the
+-- container, otherwise an exception is raised.
```

Comment:
```suggestion
-- container, otherwise a spectacle exception is raised.
```

... take it or leave it. The context makes it a little obvious I think but it might be nice to clarify for some readers that these are spectacle exceptions.